aether-producer/producer/__init__.py

```diff
@@ -268,12 +278,31 @@ def serve(self):
         )
         self.http.start()
 
+    # Basic Auth implementation
+
+    def check_auth(self, username, password):
+        return username == self.admin_name and password == self.admin_password
+
+    def authenticate(self):
```

Comment:
This name is confusing to me, do you pretend to *authenticate*? or to throw the missing authentication error response?

Reply:
If you're on a browser, I believe this response is what prompts you to authenticate with the basic auth pop-up. Maybe `request_authentication` is more clear?